File: backend/app/developer/developer_commands.py
from __future__ import annotations

from typing import Any

from app.memory.memory_store import remember
from app.world.world_state import clamp


def apply_developer_command(runtime: Any, command: dict[str, Any]) -> dict[str, Any]:
    """执行开发者通道命令。"""
    world = runtime.world
    event_store = runtime.event_store
    command_type = command.get("type")
    if command_type == "pause":
        world["clock"]["paused"] = True
    elif command_type == "resume":
        world["clock"]["paused"] = False
    elif command_type == "focus":
        world["activeFocus"] = command.get("agentId")
    elif command_type == "injectEvent":
        event_store.append(command.get("eventType") or "developer.injected", {"message": command.get("message") or "开发者注入了一个实验事件。", "payload": command.get("payload") or {}}, {"actor": "developer"})
    elif command_type == "adjustAgent" and command.get("agentId") in world["agents"]:
        agent = world["agents"][command["agentId"]]
        for key, value in (command.get("status") or {}).items():
            agent["status"][key] = clamp(int(value), 0, 999)
        remember(agent, command.get("memory") or "开发者调整了我的实验状态。", tick=world["clock"]["tick"], importance=0.75, tags=["developer"])
        event_store.append("developer.adjust_agent", {"agentId": agent["id"], "status": agent["status"]})
    return runtime.get_public_state()

File: legacy_backend_js/developer/developerCommands.js
import { remember } from '../memory/memoryStore.js';
import { clamp } from '../world/worldState.js';

export function applyDeveloperCommand(runtime, command = {}) {
  const { world, eventStore } = runtime;
  const type = command.type;
  if (type === 'pause') world.clock.paused = true;
  if (type === 'resume') world.clock.paused = false;
  if (type === 'focus') world.activeFocus = command.agentId ?? null;
  if (type === 'injectEvent') {
    eventStore.append(command.eventType ?? 'developer.injected', { message: command.message ?? '开发者注入了一个实验事件。', payload: command.payload ?? {} }, { actor: 'developer' });
  }
  if (type === 'adjustAgent' && world.agents[command.agentId]) {
    const agent = world.agents[command.agentId];
    for (const [key, value] of Object.entries(command.status ?? {})) agent.status[key] = clamp(Number(value), 0, 999);
    remember(agent, command.memory ?? '开发者调整了我的实验状态。', { tick: world.clock.tick, importance: 0.75, tags: ['developer'] });
    eventStore.append('developer.adjust_agent', { agentId: agent.id, status: agent.status });
  }
  return runtime.getPublicState();
}
